examples/dynamic_rendering/source/dynamic_rendering.cpp

```diff
@@ -0,0 +1,253 @@
+			).with_layout_transition(avk::layout::undefined >> avk::layout::color_attachment_optimal)
+		);
+
+		auto const windowResolution = mainWnd->resolution();
+		// Set the dynamic viewport only once at the start if we want shared fullscreen viewport for each renderpass
+		if(mFullscreenViewport) {
+			auto const currentViewport = vk::Viewport(0.0f, 0.0f, windowResolution.x, windowResolution.y);
+			renderCommands.emplace_back(avk::command::custom_commands([=](avk::command_buffer_t& cb) {
+				cb.handle().setViewport(0, 1, &currentViewport);
+			}));
+		} 
+
+		// We render the image in four draws - each draw has it's own clear value as well as offset and extent.
+		// Note that the renderpass extent acts as "clip" on top of the pipeline viewport.
+		// When we set the fullscreenViewport we thus get a single triangle just drawn in four parts
+		// Otherwise the viewport is set to be the current extent of the renderpass and we get four separate triangles
+		for(uint32_t drawIdx = 0; drawIdx < 4; drawIdx++)
+		{
+			float xCoord = drawIdx % 2;
+			float yCoord = drawIdx / 2u; //NOLINT(bugprone-integer-division) Standard way of unpacking 2d y coordinate from 1d encoding
+			vk::Offset2D const currOffset{
+				static_cast<int32_t>(std::min(static_cast<uint32_t>(xCoord * mXSplit + xCoord * mBorderThickness), windowResolution.x)),
+				static_cast<int32_t>(std::min(static_cast<uint32_t>(yCoord * mYSplit + yCoord * mBorderThickness), windowResolution.y))
+			};
+			vk::Extent2D const currExtent{
+				static_cast<uint32_t>(std::max(static_cast<int32_t>((1 - xCoord) * mXSplit - mBorderThickness + xCoord * (windowResolution.x - mXSplit)), 0)),
+				static_cast<uint32_t>(std::max(static_cast<int32_t>((1 - yCoord) * mYSplit - mBorderThickness + yCoord * (windowResolution.y - mYSplit)), 0)),
+			};
+			if(currExtent.height == 0 || currExtent.width == 0)
+			{
+				break;
+			}
+			colorAttachment.set_clear_color(mClearColors.at(drawIdx));
+			// Set the dynamic viewport to be equal to each of the renderpass extent and offsets if don't want shared fullscreen viewport for all of them
+			renderCommands.emplace_back(avk::command::begin_dynamic_rendering({colorAttachment}, {swapchainImageView}, currOffset, currExtent));
+			renderCommands.emplace_back(avk::command::bind_pipeline(mPipeline.as_reference()));
+			if(!mFullscreenViewport)
+			{
+				auto const currentViewport = vk::Viewport(currOffset.x, currOffset.y, currExtent.width, currExtent.height);
+				renderCommands.emplace_back(avk::command::custom_commands([=](avk::command_buffer_t& cb) {
+					cb.handle().setViewport(0, 1, &currentViewport);
+				}));
+			}
+			renderCommands.emplace_back(avk::command::draw(3u, 1u, 0u, 0u));
+			renderCommands.emplace_back(avk::command::end_dynamic_rendering());
+		}
+
+		avk::context().record(renderCommands)
+		.into_command_buffer(cmdBfr[0])
+		.then_submit_to(*mQueue)
+		.waiting_for(imageAvailableSemaphore >> avk::stage::color_attachment_output)
+		.submit();
```

Comment:
```suggestion
		avk::context().record(renderCommands)
			.into_command_buffer(cmdBfr[0])
			.then_submit_to(*mQueue)
			.waiting_for(imageAvailableSemaphore >> avk::stage::color_attachment_output)
			.submit();
```

just whitespaces

Reply:
Fixed in commit [c0b455a](https://github.com/cg-tuwien/Auto-Vk-Toolkit/pull/186/commits/c0b455af34e4c2195018acb0b3e713b844286031)